doc/onboarding.md

```diff
@@ -0,0 +1,156 @@
+## onboarding to nodejs
+
+### **thank you** for doing this
+  * collaborators are effectively part owners
+  * going to cover four things:
+    * some project goals & values
+    * local setup
+    * issues, labels, and reviewing code
+    * merging code
+
+### a little deeper about the project
+  * effectively has the goals of it's contributors
+  * but, there are some higher-level goals and values
+    * not everything belongs in core (if it can be done reasonably in userland, let is stay in userland)
+    * empathy towards users matters (this is in part why we onboard people)
+
+### setup:
+  * notifications setup
+    * use https://github.com/notifications or set up email
+    * watching the main repo will flood your inbox, so be prepared
+  * CI testing:
+    * lives here: https://ci.nodejs.org/
+    * make sure to log in – I've sent your emails to rvagg, who will set up an account for you
+    * go to "Build with parameters"
+    * fill in user, repo, and branch. don't need to touch nodes_subset.
+  * git:
+    * make sure you have whitespace=fix: `git config --global --add core.whitespace fix`
+    * fork the repo to your user account on github
+    * `git clone git://github.com/USERNAME/node.git`
+    * `git remote add upstream git://github.com/nodejs/node.git`
+    * to update from nodejs/node:
+    * `git checkout master`
+    * `git remote update -p` OR `git fetch --all` (I prefer the former)
+    * `git merge --ff-only upstream/master` (or `REMOTENAME/BRANCH`)
```

Comment:
I just do `git pull --rebase` FWIW.  Maybe you can include a short blurb on how to make git check out pull requests automatically?
```
# in .git/config
[remote "origin"]
    fetch = +refs/pr/*/head:refs/remotes/origin/pr/*
    fetch = +refs/tags/*:refs/tags/*
```
(I threw in tags for good measure.)